### PR TITLE
Workaround for bsc#1222411

### PR DESCRIPTION
**Pull request by alice-suse**
- This PR mainly does below. Now with these, in OSD, kermit can finish SL Micro 6 bare metal installation successfully.
   - usb selection on vh082: exclude usb with key word `Virtual_NetFile`
   - installed os persistent boot setting for kermit: kermit can't set the newly installed `sl-secureboot` as first boot, so do it in combustion
   - add code to pass bootparam to firstboot
   - recover default selinux config by removing customized config file
   - workaround https://bugzilla.suse.com/show_bug.cgi?id=1222411#c9

The 2 OSD machines selected, kermit and scooter , have been fully enabled and updated in openqa worker settings to do SL Micro 6 test. See https://gitlab.suse.de/openqa/salt-pillars-openqa/-/merge_requests/775. But for general sles15sp6 test, please wait for Julie to finish uefi sles verifications in poo#153037.  After https://gitlab.suse.de/openqa/salt-pillars-openqa/-/merge_requests/775 is merged, kermit can serve official SL Micro 6 test directly. Scooter needs to wait until Julie's ticket work is done. 

- Related ticket: https://progress.opensuse.org/issues/151498
- Verification run: https://openqa.suse.de/tests/14014368 : PASS

### CODE REVIEW

**Comment by alice-suse** [comment]
@waynechen55 @nanzhg @Julie-CAO  Welcome to review!

File: tests/installation/usb_install.pm [hunk 1]
```diff
@@ -29,7 +29,9 @@ sub run {
     assert_script_run("set -o pipefail");
 
     # Specify the two usb drives to use
-    my @usb = split('\n', script_output("ls /dev/disk/by-id/ -l | grep -i usb | grep -i -v -E \"generic|part\" | sed 's#^.*\\\/##'"));
+    my $cmd = "ls /dev/disk/by-id/ -l | grep -i usb | "
+      . "grep -i -v -E \"generic|part|Virtual_NetFile\" | sed 's#^.*\\\/##'";
```
Comment (waynechen55) [inline tests/installation/usb_install.pm:33]: Why have to exclude virtual_netfile ?

Comment (alice-suse) [reply tests/installation/usb_install.pm:33]: That's empty usb device, just like those with `generic ...`

Comment (nanzhg) [reply tests/installation/usb_install.pm:33]: What I observed on vh081/082 is that not only one virtual devices listed, so you have to change the keyword to filter more Virtual devices.
```
Linux_Virtual_NetFile
Linux_Virtual_CDROM
```

Comment (alice-suse) [reply tests/installation/usb_install.pm]: Updated.

**Comment by waynechen55** [comment]
Could you help show what needle ```pass-bootparam-to-firstboot``` looks like ? And how the test proceed if it is found ???

**Comment by waynechen55** [comment]
> Could you help show what needle `pass-bootparam-to-firstboot` looks like ? And how the test proceed if it is found ???

I saw it. It is here. https://openqa.suse.de/tests/14014368#step/bootloader_uefi/26

**Comment by waynechen55** [comment]
> > Could you help show what needle `pass-bootparam-to-firstboot` looks like ? And how the test proceed if it is found ???
> 
> I saw it. It is here. https://openqa.suse.de/tests/14014368#step/bootloader_uefi/26

This new rd.kiwi.install.pass.bootparam is bit longer, there is possibility it spans two lines. See:
rd_pass_bootparam_span_two_lines

**Comment by alice-suse** [comment]
> > > Could you help show what needle `pass-bootparam-to-firstboot` looks like ? And how the test proceed if it is found ???
> > 
> > 
> > I saw it. It is here. https://openqa.suse.de/tests/14014368#step/bootloader_uefi/26
> 
> This new rd.kiwi.install.pass.bootparam is bit longer, there is possibility it spans two lines. See: rd_pass_bootparam_span_two_lines

Looks like a new needle is needed.

**Comment by alice-suse** [comment]
The openqa worker kermit and scooter [MR](https://gitlab.suse.de/openqa/salt-pillars-openqa/-/merge_requests/775) for usb install, is merged.

Merging this PR, will make OSD tests workable in [development group](https://openqa.suse.de/group_overview/544). Can we merge this PR now?

**Comment by waynechen55** [comment]
> > > > Could you help show what needle `pass-bootparam-to-firstboot` looks like ? And how the test proceed if it is found ???
> > > 
> > > 
> > > I saw it. It is here. https://openqa.suse.de/tests/14014368#step/bootloader_uefi/26
> > 
> > 
> > This new rd.kiwi.install.pass.bootparam is bit longer, there is possibility it spans two lines.
> 
> Looks like a new needle is needed.

A new needle can solve the problem because this line is too long and its position is unpredictable.

**Comment by alice-suse** [comment]
> > > This new rd.kiwi.install.pass.bootparam is bit longer, there is possibility it spans two lines.
> > 
> > 
> > Looks like a new needle is needed.
> 
> A new needle can not solve the problem because this line is too long and its position is unpredictable.

This parameter is defined by kiwi. We can't make it shorter. 
When it spans two lines, adding needle with two parts selected per the two lines, shall be fine, I think. Its position can differ for vh015 and kermit. But catching needles for each of the machine shall be fine, right? We do not need to have the test run on every ipmi machine, but only selected machines. And for each machine, it shall be fixed needle based on my observations. I thought about why needles of this differ. I think it is because we can't set the video when booting from iso, which is different from pxe boot when we can set fixed value.

**Comment by waynechen55** [comment]
I think merging this pull request is OK at the moment.

**Comment by alice-suse** [comment]
> I think merging this pull request is OK at the moment.

There was a new build yesterday in OSD. Our tests can run with this PR merged.

**Pull request merged**